src/RelayPricer.ts

```diff
@@ -0,0 +1,42 @@
+import CoinBase from './api/CoinBase';
+import { SourceApi } from './types/RelayPricer';
+
+export default class RelayPricer {
+    private readonly sourceApi: SourceApi;
+
+    constructor(sourceApi: SourceApi | string) {
+        if (typeof sourceApi === 'string' || sourceApi instanceof String) {
+            switch (sourceApi as string) {
+                case 'coinbase':
+                    this.sourceApi = new CoinBase();
+                    break;
+                default:
+                    throw new Error('Source api not provided');
+            }
+        } else {
+            this.sourceApi = sourceApi as SourceApi;
+        }
+    }
+
+    async getPrice(
+        sourceCurrency: string,
+        targetCurrency: string,
+        intermediateCurrency?: string
+    ): Promise<number> {
+        const intermediary = intermediateCurrency
+            ? intermediateCurrency
+            : 'USD';
+        const sourcePrice = await this.sourceApi.query(
+            sourceCurrency,
+            intermediary
+        );
+        const targetPrice = await this.sourceApi.query(
+            targetCurrency,
+            intermediary
+        );
```

Comment:
I think that those to calls can be performed in parallel

Reply:
updated